core/Network/TLS/Handshake/Client.hs

```diff
@@ -483,19 +483,24 @@ sendClientData cparams ctx = sendCertificate >> sendClientKeyXchg >> sendCertifi
                         ffGroup = findFiniteFieldGroup params
                         srvpub  = serverDHParamsToPublic serverParams
 
+                    unless (maybe False (isSupportedGroup ctx) ffGroup) $ do
+                        groupUsage <- (onCustomFFDHEGroup $ clientHooks cparams) params srvpub `catchException`
```

Comment:
Minor issue: this should be `onCustomFFDHEGroup (clientHooks cparams)`.
I will take care of this.